packages/data/src/user/setProfileInfo.ts

```diff
@@ -0,0 +1,47 @@
+import { Client as FaunaDBClient, query as q } from "faunadb";
+import { DatabaseUser } from "./databaseUser";
+
+export const setProfileInfo = (client: FaunaDBClient) => {
+    return async (id: string, nickname: string, picture: number) => {
+        try {
+            let user;
+            if (nickname === null) {
```

Comment:
code style 💅: As this has access to both `nickname` and `picture`, it would be clearer if they were both then put into a single object.

```ts
let userUpdate = {}

if (nickname) {
    userUpdate = {
        ...userUpdate,
        nickname: {
            value: nickname,
            // todo do this in the db?
            uppercase: nickname.toUpperCase()
        }
    }
}

if (picture) {
    userUpdate = {
        ...userUpdate,
        profile: {
            picture
        }
    }
}
```

and then use that in the query.